include/lemlib/chassis/differential.hpp

```diff
@@ -294,16 +298,16 @@ class Differential : public Chassis {
          * curve, refer to the `defaultDriveCurve` documentation.
          */
         void curvature(int throttle, int turn, float cureGain = 0.0,
-                       DriveCurveFunction_t driveCurve = defaultDriveCurve);
+                       const DriveCurveFunction_t& driveCurve = defaultDriveCurve);
     private:
         /**
          * @brief Chassis update function. Updates chassis motion and odometry
          *
          */
         void update() override;
 
-        ControllerSettings linearSettings;
-        ControllerSettings angularSettings;
-        Drivetrain drivetrain;
+        std::unique_ptr<ControllerSettings> linearSettings;
+        std::unique_ptr<ControllerSettings> angularSettings;
+        std::unique_ptr<Drivetrain> drivetrain;
```

Comment:
Why do these need to be pointers?

Reply:
these are the advantages:
1. once differential objects got destroyed, they went out of scope and then destroyed themselves
2. compare . vs ->, -> is easier to read(this is one of an reason why ppl use pointer)
3. the differential will have the ownership of these instances